### PR TITLE
remove service user from get acls

### PR DESCRIPTION
**Pull request by Tianhao-Gu**
remove users associated with service tokens from the get_samples_acls method
https://kbase-jira.atlassian.net/browse/SCT-3079

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
@slebras Do you mind taking a look?

File: lib/SampleService/core/api_translation.py [hunk 1]
```diff
@@ -36,6 +36,8 @@
 
 ID = 'id'
 ''' The ID of a sample. '''
+SERVICE_USER = ['indexer_user', 'jayrbolton']
```
Comment (scanon) [inline lib/SampleService/core/api_translation.py:39]: This needs to be a configurable that can be specified in the config file.

**Review by scanon — CHANGES_REQUESTED** [review]
I think this looks good but just need to make the SERVICE_USER configurable.

**Comment by MrCreosote** [comment]
this seems like a crazy hack. Why not just not put the service users in the acl in the first place?

**Comment by scanon** [comment]
> this seems like a crazy hack. Why not just not put the service users in the acl in the first place?

When the indexer reads the sampleset object (so it can index it), the workspace then grants the indexer read permission to the samples.  So this is what we are trying to mask.  While we could try to make changes to the workspace to not do the ACL update in some scenarios, this seemed like the easiest solution.

**Comment by MrCreosote** [comment]
I see. An alternative workaround would be to have the indexer remove itself from the acl once it's done, so you remove the ACL pollution rather than mask it.

**Comment by MrCreosote** [comment]
But wait a sec - why is the indexer triggering based on sample set objects? Why doesn't it just trigger on the kafka messages from the sample service?

**Comment by scanon** [comment]
> But wait a sec - why is the indexer triggering based on sample set objects? Why doesn't it just trigger on the kafka messages from the sample service?

It will trigger on both.  But the issue here is it gets triggered on the sample set and when it reads the object data that triggers the sample acl update.  We could have it drop the ACL after it is done but I'm not convinced that is a great option either.  I think this will work okay for now.  We are going to switch the indexers to use a special account so we can safely mask it since it shouldn't be used for general use.

File: lib/SampleService/core/api_translation.py (hunk 1 above)
Comment (scanon) [reply lib/SampleService/core/api_translation.py]: This is addressed now.

**Review by scanon — APPROVED** [review]
LGTM

**Comment by scanon** [comment]
We are going to merge this.  Long-term we want to look at some better alternatives since this is a bit of a hack.  But this seems acceptable for now.

**Pull request merged**